### ISSUE TITLE
fb_client.fetchThreadMessages raises FBchatFacebookError: Error when sending request: Got 500 response

### ISSUE THREAD

**Issue by xbonio** [carpedm20/fbchat#503]
I'm trying to scrape a conversation of ~20k messages and I'm getting `fbchat._exception.FBchatFacebookError: Error when sending request: Got 500 response` with below reproduction.

## Code to reproduce
```py
fb_client.fetchThreadMessages('123456789', limit=20000)
```

## Traceback
```
Traceback (most recent call last):
  File "fbm-scraper.py", line 105, in <module>
    messages = fb_client.fetchThreadMessages('1234567890', limit=20000)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_client.py", line 783, in fetchThreadMessages
    j = self.graphql_request(_graphql.from_doc_id("1860982147341344", params))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_client.py", line 185, in graphql_request
    return self.graphql_requests(query)[0]
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_client.py", line 177, in graphql_requests
    return tuple(self._post("/api/graphqlbatch/", data, as_graphql=True))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_client.py", line 134, in _post
    content = check_request(r)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_util.py", line 156, in check_request
    check_http_code(r.status_code)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fbchat/_util.py", line 171, in check_http_code
    raise FBchatFacebookError(msg, request_status_code=code)
fbchat._exception.FBchatFacebookError: Error when sending request: Got 500 response.
```

## Environment information
- Reproduces both on py3.7 and py3.8
- `fbchat` version: 1.9.3

**Comment by madsmtm** [comment]
Thanks for the report, though this is not _really_ an issue on our side.

The issue is probably that Facebook allow you to request 20000 records simultaneously in the first place. Then, their servers probably has some sort of "max request time" limit, which kicks in, and kills the request (hence the 500 error).

In the future, `fbchat` should handle these "max limits" for you automatically, but for now, you can split the request into pieces, by doing something like this:
```python
BATCH_SIZE = 2000
all_messages = []
before = None
while True:
    messages = fb_client.fetchThreadMessages("123456789", limit=BATCH_SIZE, before=before)
    if not messages:  # If there were no messages returned, we're done
        break
    all_messages.extend(messages)
    # We may lose some messages here if they were sent in the same millisecond
    before = int(messages[-1].timestamp) - 1
```

**Comment by madsmtm** [comment]
This is fixed in `v2`

**Comment by simonayzman** [comment]
@madsmtm Would love to use the automatic batching functionality. I looked at the Project status so far. Is there a rough estimate of when v2 might get released? Worst case, I can use bleeding edge `master`, but would rather not if I can help it.

**Comment by madsmtm** [comment]
Can't really give you an estimate, depends a lot of when I have the time. But I've just released an alpha version `v2.0.0a1` that you can play with. It is fairly "stable", in the sense that it doesn't crash, but there will be breaking changes between alpha versions!

**Comment by simonayzman** [comment]
I appreciate it! Thanks @madsmtm
